packages/asset-swapper/src/utils/alt_mm_implementation_utils.ts

```diff
@@ -0,0 +1,236 @@
+        ) {
+            return offering;
+        }
+    }
+    return undefined;
+}
+
+function parseFirmQuoteResponseFromAltMM(altFirmQuoteReponse: AltFirmQuoteResponse): V4RFQFirmQuote {
+    return {
+        signedOrder: altFirmQuoteReponse.data['0xv4order'],
+    };
+}
+
+function parseIndicativeQuoteResponseFromAltMM(
+    altIndicativeQuoteResponse: AltIndicativeQuoteResponse,
+    altPair: AltOffering,
+    makerToken: string,
+    takerToken: string,
+): V4RFQIndicativeQuote {
+    let makerAmount: BigNumber;
+    let takerAmount: BigNumber;
+    let quoteAmount: BigNumber;
+    let baseAmount: BigNumber;
+
+    if (!altIndicativeQuoteResponse.price) {
+        throw new Error('Price not returned by alt MM');
+    }
+    if (altIndicativeQuoteResponse.amount) {
+        // if amount is specified, amount is the base token amount
+        baseAmount = Web3Wrapper.toBaseUnitAmount(
+            new BigNumber(altIndicativeQuoteResponse.amount),
+            altPair.baseAssetDecimals,
+        );
+        // if amount is specified, use the price (quote/base) to get the quote amount
+        quoteAmount = Web3Wrapper.toBaseUnitAmount(
+            new BigNumber(altIndicativeQuoteResponse.amount)
+                .times(new BigNumber(altIndicativeQuoteResponse.price))
+                .decimalPlaces(altPair.quoteAssetDecimals, BigNumber.ROUND_DOWN),
+            altPair.quoteAssetDecimals,
+        );
+    } else if (altIndicativeQuoteResponse.value) {
+        // if value is specified, value is the quote token amount
+        quoteAmount = Web3Wrapper.toBaseUnitAmount(
+            new BigNumber(altIndicativeQuoteResponse.value),
+            altPair.quoteAssetDecimals,
+        );
+        // if value is specified, use the price (quote/base) to get the base amount
+        baseAmount = Web3Wrapper.toBaseUnitAmount(
+            new BigNumber(altIndicativeQuoteResponse.value)
+                .dividedBy(new BigNumber(altIndicativeQuoteResponse.price))
+                .decimalPlaces(altPair.baseAssetDecimals, BigNumber.ROUND_DOWN),
+            altPair.baseAssetDecimals,
+        );
+    } else {
+        throw new Error('neither amount or value were specified');
+    }
+    if (makerToken.toLowerCase() === altPair.baseAsset.toLowerCase()) {
+        makerAmount = baseAmount;
+        takerAmount = quoteAmount;
+    } else if (makerToken.toLowerCase() === altPair.quoteAsset.toLowerCase()) {
+        makerAmount = quoteAmount;
+        takerAmount = baseAmount;
+    } else {
+        throw new Error(`Base, quote tokens don't align with maker, taker tokens`);
+    }
+
+    return {
+        makerToken,
+        makerAmount,
+        takerToken,
+        takerAmount,
+        // HACK: alt implementation does not return an expiration with indicative quotes
+        // return now + { IMPUTED EXPIRY SECONDS } to have it included after order checks
+        expiry: new BigNumber(
```

Comment:
Do we need to wrap this BigNumber in another BigNumber?

Reply:
nope 👍